server/src/main/java/org/cloudfoundry/identity/uaa/zone/VmidentityIdentityZoneProvisioning.java

```diff
@@ -79,42 +79,84 @@ public IdentityZone retrieveBySubdomain(String subdomain) {
     @Override
     public List<IdentityZone> retrieveAll() {
         try {
-            String systemTenant = this._idmClient.getSystemTenant();
             ArrayList<IdentityZone> list = new ArrayList<IdentityZone>();
             for (String tenant : this._idmClient.getAllTenants()) {
-                list.add(identityZoneForTenant(this._idmClient.getTenant(tenant),
-                        systemTenant.equalsIgnoreCase(tenant)));
+                list.add(getIdentityZoneFromTenant(tenant));
             }
-
+            
             return list;
         } catch (Exception ex) {
             throw new IllegalStateException("Failed to retrieve zones.", ex);
         }
     }
+    
+    private IdentityZone identityZoneForTenant(String tenantName) throws Exception {
+        IdentityZone zone = new IdentityZone();
+        Tenant t = this._idmClient.getTenant(tenantName);
+        
+        zone.setId(t.getName());
+        zone.setName(t.getName());
+        zone.setSubdomain(t.getName());
+        zone.setDescription(t._longName);
+        zone.setVersion(1);
+        
+        return zone;
+    }
+    
+    private void configureIdentityZone(String tenantName, IdentityZone zone) throws Exception {
+        IdentityZoneConfiguration config = this._configs.get(tenantName);
```

Comment:
They're not fully wired - updates are unavailable, so at least some of it still get stored only in memory.